### PR TITLE
hobbits: init at 0.52.0

### PR DESCRIPTION
**Pull request by sikmir**
#### Motivation for this change
[Hobbits](https://github.com/Mahlet-Inc/hobbits) - A multi-platform GUI for bit-based analysis, processing, and visualization.

###### Things done
- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by privateseabass** [comment]
This built for me.
Not sure how it works, but the GUI started.

**Comment by sikmir** [comment]
Keep it.

File: pkgs/tools/graphics/hobbits/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{ stdenv, mkDerivation, fetchFromGitHub, makeDesktopItem, qmake, python3
+, withAutocorrelation ? true, fftw, pkgconfig  }:
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/hobbits/default.nix:2]:
```suggestion
{ stdenv, mkDerivation, fetchFromGitHub, makeDesktopItem, qmake, python3, fftw
, withAutocorrelation ? true, pkg-config  }:
```

File: pkgs/tools/graphics/hobbits/default.nix [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  desktopItem = makeDesktopItem {
+    name = "hobbits";
+    exec = "hobbits";
+    icon = "hobbits";
+    desktopName = "Hobbits GUI";
+  };
+
+  buildInputs = stdenv.lib.optional withAutocorrelation fftw;
+  nativeBuildInputs = [ qmake ] ++ stdenv.lib.optional withAutocorrelation pkgconfig;
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/hobbits/default.nix:25]:
```suggestion
  nativeBuildInputs = [ qmake ] ++ stdenv.lib.optional withAutocorrelation pkg-config;
```

File: pkgs/tools/graphics/hobbits/default.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  qmakeFlags = [
+    ''DEFINES+=HOBBITS_GUI_VERSION=\"\\\"${version}\\\"\"''
+    ''DEFINES+=HOBBITS_CORE_LIB_VERSION=\"\\\"${version}\\\"\"''
+    ''DEFINES+=HOBBITS_RUNNER_VERSION=\"\\\"${version}\\\"\"''
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/hobbits/default.nix:30]:
```suggestion
    "DEFINES+=HOBBITS_GUI_VERSION=\"version\""
    "DEFINES+=HOBBITS_CORE_LIB_VERSION=\"version\""
    "DEFINES+=HOBBITS_RUNNER_VERSION=\"version\""
```

File: pkgs/tools/graphics/hobbits/default.nix [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/Mahlet-Inc/hobbits";
+    description = "A multi-platform GUI for bit-based analysis, processing, and visualization";
+    license = licenses.mit;
+    maintainers = with maintainers; [ sikmir ];
+    platforms = with platforms; linux;
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/hobbits/default.nix:71]:
```suggestion
    platforms = with platforms.linux;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Does not build:

```
peakselector.cpp: In member function 'virtual void PeakSelector::paintEvent(QPaintEvent*)':
peakselector.cpp:69:18: error: aggregate 'QPainterPath path' has incomplete type and cannot be defined
   69 |     QPainterPath path;
      |                  ^~~~

```

**Comment by sikmir** [comment]
0.50.0 is out, there are a lot of changes since 0.23.0.

**Pull request merged**